operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -46,30 +45,75 @@ const (
 // This type exists so there's no chance of pushing random strings to our progress status
 type ProgressState string
 
+const (
+	defaultConfigBuilderImage     = "datastax/cass-config-builder:1.0.0"
+	cassandra_3_11_6              = "datastax/cassandra-mgmtapi-3_11_6:v0.1.5"
+	cassandra_4_0_0               = "datastax/cassandra-mgmtapi-4_0_0:v0.1.5"
+	dse_6_8_0                     = "datastax/dse-server:6.8.0"
+	ubi_cassandra_3_11_6          = "datastax/cassandra:3.11.6-ubi7"
+	ubi_cassandra_4_0_0           = "datastax/cassandra:4.0-ubi7"
+	ubi_dse_6_8_0                 = "datastax/dse-server:6.8.0-ubi7"
+	ubi_defaultConfigBuilderImage = "datastax/cass-config-builder:1.0.0-ubi7"
+	EnvBaseImageOs                = "BASE_IMAGE_OS"
+)
+
 // getImageForServerVersion tries to look up a known image for a server type and version number.
 // In the event that no image is found, an error is returned
 func getImageForServerVersion(server, version string) (string, error) {
-	const (
-		cassandra_3_11_6 = "datastax/cassandra-mgmtapi-3_11_6:v0.1.5"
-		cassandra_4_0_0  = "datastax/cassandra-mgmtapi-4_0_0:v0.1.5"
-		dse_6_8_0        = "datastax/dse-server:6.8.0"
-	)
-	sv := server + "-" + version
+	baseImageOs := os.Getenv(EnvBaseImageOs)
+
+	var imageCalc func(string) (string, bool)
+	var img string
+	var success bool
+	var errMsg string
+
+	if baseImageOs == "" {
+		imageCalc = getImageForDefaultBaseOs
+		errMsg = fmt.Sprintf("server '%s' and version '%s' do not work together", server, version)
+	} else {
+		// if this operator was compiled using a UBI base image
+		// such as registry.access.redhat.com/ubi7/ubi-minimal:7.8
+		// then we use specific cassandra and init container coordinates
+		// that are built accordingly
+		errMsg = fmt.Sprintf("server '%s' and version '%s', along with the specified base OS '%s', do not work together", server, version, baseImageOs)
+		imageCalc = getImageForUniversalBaseOs
+	}
+
+	img, success = imageCalc(server + "-" + version)
+	if !success {
+		return "", fmt.Errorf(errMsg)
+	}
+
+	return img, nil
+}
+
+func getImageForDefaultBaseOs(sv string) (string, bool) {
+	switch sv {
+	case "dse-6.8.0":
+		return dse_6_8_0, true
+	case "cassandra-3.11.6":
+		return cassandra_3_11_6, true
+	case "cassandra-4.0.0":
+		return cassandra_4_0_0, true
+	}
+	return "", false
+}
+
+func getImageForUniversalBaseOs(sv string) (string, bool) {
 	switch sv {
 	case "dse-6.8.0":
```

Comment:
What if we turn this into a map somewhere at the top of the file, so we don't have to dig all the way down into these functions to update it?

Reply:
I think we have plans to update some of this for 6.8.1 support in a different PR.